proxy/logging/LogAccess.cc

```diff
@@ -1298,6 +1298,44 @@ LogAccess::marshal_client_sni_server_name(char *buf)
   return len;
 }
 
+/*-------------------------------------------------------------------------
+  -------------------------------------------------------------------------*/
+int
+LogAccess::marshal_client_provided_cert(char *buf)
+{
+  int provided_cert = 0;
+  if (m_http_sm) {
+    auto txn = m_http_sm->get_ua_txn();
+    if (txn) {
+      auto ssn = txn->get_proxy_ssn();
+      if (ssn) {
+        auto ssl = ssn->ssl();
+        if (ssl) {
+          provided_cert = ssl->client_provided_certificate();
```

Comment:
In the past, code similar to this has caused segment violations, due to race conditions between the destruction times of associated objects.